### ISSUE TITLE
Streams from saClient.files.getReadStream don't wait for listeners

### ISSUE THREAD

**Issue by coolaj86** [box/box-node-sdk#704]
- [x] I have checked that the [SDK documentation][sdk-docs] doesn't solve my issue.
- [x] I have checked that the [API documentation][api-docs] doesn't solve my issue.
- [x] I have searched the [Box Developer Forums][dev-forums] and my issue isn't already reported (or if it has been reported, I have attached a link to it, for reference).
- [x] I have searched [Issues in this repo][github-repo] and my issue isn't already reported.

The **SDK and API** links are **out-of-date** ^^ (but I did search best I could despite that)

### Description of the Issue

Streams from `getReadStream` are already streaming before there are any listeners.

Easiest to describe with code:

### Setup

```js
let BoxSDK = require('box-node-sdk');
let Sdk = new BoxSDK({
  clientID: process.env.BOX_CLIENT_ID,
  clientSecret: process.env.BOX_CLIENT_SECRET,
  appAuth: {
    keyID: process.env.BOX_PUBLIC_KEY_ID,
    passphrase: process.env.BOX_PRIVATE_KEY_PASSPHRASE,
    privateKey: box_private_key,
  },
});
let saClient = Sdk.getAppAuthClient('enterprise', process.env.BOX_APP_ENTERPRISE_ID);
```

### Buggy Example

```js
let fileStream = await saClient.files.getReadStream(id);

setTimeout(function () {
  // There's no data left!
  fileStream.pipe(other);
}, 1000);
```

### Workaround

```js
let fileStream = await saClient.files.getReadStream(id);
fileStream.pause();

setTimeout(function () {
  fileStream.pipe(other);
  fileStream.resume();
}, 1000);
```

### Why the Workaround doesn't work

We have several processes where there may be a space in time between when we get the box stream and when we attach it to the outbound pipes.

In each of those we have to put a hacky `fileStream.resume()` right after them.

This starts to get tricky when we have multiple outputs or need to handle the box streams sequentially rather than all-at-once.

### Versions Used

Node SDK: `"box-node-sdk": "^2.0.0",`

[sdk-docs]: ./doc
[api-docs]: https://developer.box.com/docs
[dev-forums]: https://community.box.com/t5/Platform-and-Development-Forum/bd-p/DeveloperForum
[github-repo]: https://github.com/box/box-java-sdk/search?type=Issues

**Comment by mhagmajer** [comment]
Thanks for posting @coolaj86! I'll added this to our backlog, will get back to you as we work on it.

**Comment by mhagmajer** [comment]
Hello @coolaj86!

I've done testing around this issue using Node 12+ and box-node-sdk v2.3.0. When using the following script (`sdkClient` is a basic Node SDK client authenticated with JWT):

```ts
  const fileStream = await sdkClient.files.getReadStream('969474247378');
  fileStream.pipe(process.stdout);
```

I've found that all file contents are correctly printed on screen as they should.

However, if I explicitly paused the stream after it was received, like so:

```ts
  const fileStream = await sdkClient.files.getReadStream('969474247378');
  fileStream.pause();
  fileStream.pipe(process.stdout);
```

No contents were printed and the program was stuck in running state.

I don't think that we should return streams in paused state because they'd need to be explicitly resumed before they can be consumed with pipe.

I've also tried the example that you've provided, however, I got the data in timeout handler.

I don't see anything to fix here at the moment. Please test again with the updated version of SDK and reopen the issue as you see fit. Thanks again for posting!

**Comment by coolaj86** [comment]
You didn't see the error because you didn't do a `setTimout()` to simulate asynchronous usage of the stream:

```js
let fileStream = await saClient.files.getReadStream(id);

setTimeout(function () {
  // There's no data left!
  fileStream.pipe(other);
}, 1000);
```

**Comment by coolaj86** [comment]
I'll update to the latest version of the API and see if I can still reproduce on my end.

**Comment by congminh1254** [comment]
Hi @coolaj86 ,

We did a release contains your bug fix, you can use Node SDK 2.8.1 to verify if it works for you.

Regards,
Minh
